evm/its.js

```diff
@@ -0,0 +1,723 @@
+
+            break;
+        }
+
+        case 'interchainTokenId': {
+            const sender = options.sender;
+
+            if (!isValidAddress(sender)) {
+                throw new Error(`Invalid sender address: ${sender}`);
+            }
+
+            const salt = options.salt;
+
+            if (!isKeccak256Hash(salt)) {
+                throw new Error(`Invalid salt: ${salt}`);
+            }
+
+            const interchainTokenId = await interchainTokenService.interchainTokenId(sender, salt);
+            printInfo(`InterchainTokenId for sender ${sender} and deployment salt: ${salt}`, interchainTokenId);
+
+            break;
+        }
+
+        case 'tokenManagerImplementation': {
+            const type = options.type;
+
+            const tokenManagerImplementation = await interchainTokenService.tokenManagerImplementation(tokenManagerImplementations[type]);
+            printInfo(`${type} TokenManager implementation address:`, tokenManagerImplementation);
+
+            break;
+        }
+
+        case 'flowLimit': {
+            const tokenId = options.tokenId;
+
+            if (!isValidTokenId(tokenId)) {
+                throw new Error(`Invalid tokenId value: ${tokenId}`);
+            }
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            const flowLimit = await interchainTokenService.flowLimit(tokenIdBytes32);
+            printInfo(`Flow limit for TokenManager with tokenId: ${tokenId}`, flowLimit);
+
+            break;
+        }
+
+        case 'flowOutAmount': {
+            const tokenId = options.tokenId;
+
+            if (!isValidTokenId(tokenId)) {
+                throw new Error(`Invalid tokenId value: ${tokenId}`);
+            }
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            const flowOutAmount = await interchainTokenService.flowOutAmount(tokenIdBytes32);
+            printInfo(`Flow out amount for TokenManager with tokenId: ${tokenId}`, flowOutAmount);
+
+            break;
+        }
+
+        case 'flowInAmount': {
+            const tokenId = options.tokenId;
+
+            if (!isValidTokenId(tokenId)) {
+                throw new Error(`Invalid tokenId value: ${tokenId}`);
+            }
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            const flowInAmount = await interchainTokenService.flowInAmount(tokenIdBytes32);
+            printInfo(`Flow out amount for TokenManager with tokenId: ${tokenId}`, flowInAmount);
+
+            break;
+        }
+
+        case 'deployTokenManager': {
+            const isPaused = await interchainTokenService.paused();
+
+            if (isPaused) {
+                throw new Error(`${action} invalid while service is paused.`);
+            }
+
+            const { salt, destinationChain, type, params, gasValue } = options;
```

Comment:
Shall we just do arbitrary JSON encoded args instead?

Reply:
Mind elaborating here?